### PR TITLE
feat(container): generate library

### LINKED ISSUE #7981: Kubernetes Engine API: `container.googleapis.com`

### PR DESCRIPTION
**Pull request by scotthart**
Fixes #7981 

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8109)

### CODE REVIEW

File: google/cloud/container/quickstart/quickstart.cc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+int main(int argc, char* argv[]) try {
+  if (argc != 2) {
+    std::cerr << "Usage: " << argv[0] << " project-id\n";
+    return 1;
+  }
+
+  namespace container = ::google::cloud::container;
+  auto client = container::ClusterManagerClient(
+      container::MakeClusterManagerConnection(/* EDIT HERE */));
```
Comment (dbolduc) [inline google/cloud/container/quickstart/quickstart.cc:28]: delete `/* EDIT HERE */`

File: google/cloud/container/quickstart/quickstart.cc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  auto const project = google::cloud::Project(argv[1]);
```
Comment (dbolduc) [inline google/cloud/container/quickstart/quickstart.cc:30]: looks like we want `project + "/locations/-"`

https://github.com/googleapis/google-cloud-cpp/blob/8a68ee4c1d8d16d81496dd25fb2e47b20cbf6b72/google/cloud/container/cluster_manager_client.h#L93-L96

File: google/cloud/container/quickstart/quickstart.cc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  auto response = client.ListClusters(project.FullName());
+  if (response.ok()) {
+    for (auto r : response->clusters()) {
+      std::cout << r.DebugString() << "\n";
+    }
+  } else {
+    throw std::runtime_error(response.status().message());
+  }
```
Comment (dbolduc) [inline google/cloud/container/quickstart/quickstart.cc:38]: if you want to save some lines:
```cc
if (!response) throw std::runtime_error(response.status().message());
for (auto c : response->clusters()) {
  std::cout << c.DebugString() << "\n";
}
```

**Review by dbolduc — CHANGES_REQUESTED** [review]
just some changes to the quickstart

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8109)* status: 0 of 42 files reviewed, 3 unresolved discussions (waiting on @dbolduc)

___
*[google/cloud/container/quickstart/quickstart.cc, line 28 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8109#-MuInrRp0AXhDJEmVFjW-r2-28:-MuMfyaT0pMFibEujKhq:bwohspi) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8a68ee4c1d8d16d81496dd25fb2e47b20cbf6b72/google/cloud/container/quickstart/quickstart.cc#L28)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

delete `/* EDIT HERE */`
</blockquote></details>

Thanks. I thought I had fixed the quickstart before I set this PR aside, but apparently not.
___
*[google/cloud/container/quickstart/quickstart.cc, line 30 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8109#-MuInrRp0AXhDJEmVFjW-r2-30:-MuMg98H4AvIBmywb3h5:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8a68ee4c1d8d16d81496dd25fb2e47b20cbf6b72/google/cloud/container/quickstart/quickstart.cc#L30)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

looks like we want `project + "/locations/-"`

https://github.com/googleapis/google-cloud-cpp/blob/8a68ee4c1d8d16d81496dd25fb2e47b20cbf6b72/google/cloud/container/cluster_manager_client.h#L93-L96
</blockquote></details>

Done.
___
*[google/cloud/container/quickstart/quickstart.cc, line 38 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8109#-MuInrRp0AXhDJEmVFjW-r2-38:-MuMgA5q5FQO3INkB5Kl:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8a68ee4c1d8d16d81496dd25fb2e47b20cbf6b72/google/cloud/container/quickstart/quickstart.cc#L38)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

if you want to save some lines:
```cc
if (!response) throw std::runtime_error(response.status().message());
for (auto c : response->clusters()) {
  std::cout << c.DebugString() << "\n";
}
```
</blockquote></details>

Done.

**Review by dbolduc — APPROVED** [review]

File: google/cloud/container/quickstart/quickstart.cc (hunk 3 above)
Comment (dbolduc) [reply google/cloud/container/quickstart/quickstart.cc:38]: FYI: I do not see this change, but who cares.

**Comment by coryan** [comment]
@scotthart I added a magic word so merging this PR will automatically close the corresponding bug.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8109)* status: 0 of 42 files reviewed, 3 unresolved discussions (waiting on @coryan and @dbolduc)

___
*[google/cloud/container/quickstart/quickstart.cc, line 38 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8109#-MuInrRp0AXhDJEmVFjW-r2-38:-MuMlm3WCUmNGlW1AmuM:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8a68ee4c1d8d16d81496dd25fb2e47b20cbf6b72/google/cloud/container/quickstart/quickstart.cc#L38)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

I do.  But if @scotthart would rather merge this and fix it in a follow up PR that is fine by me.
</blockquote></details>

Done.

**Comment by coryan** [comment]
@scotthart looks like you need to rebase this PR and fix the conflicts.

**Review by coryan — APPROVED** [review]
:lgtm:

> Reviewed 34 of 42 files at r1, 1 of 1 files at r2, 2 of 2 files at r4, 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8109#-:-MucAzrp9Prfyw6fHyN_:bnfp4nl)* status: all files reviewed, 3 unresolved discussions (waiting on @dbolduc)

**Comment by coryan** [comment]
@scotthart Looks like another round of rebase and regeneration is needed?

**Pull request merged**